### ISSUE TITLE
GraphRBAC ActiveDirectoryApplications.GetByName throws BadRequest

### ISSUE THREAD

**Issue by ashvins** [Azure/azure-libraries-for-net#40]
I am seeing BadRequest error thrown by Graph API when I call the Graph RBAC ActiveDirectoryApplications.GetByName() method
The Graph API error is: 

{"odata.error":{"code":"Request_UnsupportedQuery","message":{"lang":"en","value":"Unsupported or invalid query filter clause specified for property 'appId' of resource 'Application'."}}}

Looking at the code the filter query for "appId eq '<name>'" is failing. The query before it - "displayName eq '<name>'" - returned nothing.

Fluent SDK version 1.1.3

Here's the captured request:
```
GET https://graph.windows.net/<REDACTED>/applications?$filter=appId%20eq%20'<REDACTED>'&api-version=1.6 HTTP/1.1
Connection: Keep-Alive
Accept-Encoding: gzip, deflate
Accept-Language: en-US
Authorization: Bearer <REDACTED>
User-Agent: FxVersion/4.6.25211.01 Microsoft.Azure.Management.Graph.RBAC.Fluent.GraphRbacManagementClient/1.1.3.0 MacAddressHash/f3a75dfd19a09371224b56bbba4d09aee55396e000a67ef307b07b7d651172ee
x-ms-client-request-id: 9985b3d0-30f7-4b87-adc8-8a6f3739d01b
Host: graph.windows.net
```

Just figured that this does not happen when the app name is a GUID. Not sure if this is a Fluent SDK bug or AAD issue.

**Comment by ashvins** [comment]
Hello - Any update?

**Comment by andersjoh** [comment]
Same issue here.

**Comment by hovsepm** [comment]
Fixed in version 1.4
